engine/src/main/scala/cromwell/engine/workflow/lifecycle/WorkflowExecutionActor.scala

```diff
@@ -105,58 +110,56 @@ final case class WorkflowExecutionActor(workflowId: WorkflowId, workflowDescript
   val tag = self.path.name
 
   private val calls = workflowDescriptor.backendDescriptor.workflowNamespace.workflow.calls
+  private val inputResolver = new JobInputEvaluator(workflowDescriptor)
 
   // Initialize the StateData with ExecutionStore (all calls as NotStarted) and SymbolStore
   startWith(
     WorkflowExecutionPendingState,
     WorkflowExecutionActorData(
       executionStore = (calls map (BackendJobDescriptorKey(_, None, 1) -> NotStarted)) toMap,
-      symbolCache = buildSymbolStoreEntries.groupBy(entry => SymbolCacheKey(entry.scope, entry.isInput))))
+      outputStore = Map.empty))
 
   /** PBE: the return value of WorkflowExecutionActorState is just temporary.
     *      This should probably return a Try[BackendJobDescriptor], Unit, Boolean,
     *      Try[ActorRef], or something to indicate if the job was started
     *      successfully.  Or, if it can fail to start, some indication of why it
     *      failed to start
     */
-  private def startJob(jobKey: BackendJobDescriptorKey,
-                       symbolCache: SymbolCache,
+  private def executeJob(jobKey: BackendJobDescriptorKey,
+                       outputStore: OutputStore,
                        configDescriptor: BackendConfigurationDescriptor,
                        factory: BackendLifecycleActorFactory): Try[ExecutionStatus] = {
-
-      val call = jobKey.call
-      fetchLocallyQualifiedInputs(jobKey, symbolCache) match {
-        case Success(symbolStoreForCall) =>
-          val jobDescriptor = BackendJobDescriptor(workflowDescriptor.backendDescriptor, jobKey, symbolStoreForCall)
-          val actorName = s"${jobDescriptor.descriptor.id}-BackendExecutionActor-${jobKey.call.fullyQualifiedName}-${jobKey.index.getOrElse("")}-${jobKey.attempt}"
-          val jobExecutionActor = context.actorOf(
-            factory.jobExecutionActorProps(
-              jobDescriptor,
-              BackendConfigurationDescriptor(configDescriptor.backendConfig, configDescriptor.globalConfig)
-            ),
-            actorName
-          )
-          jobExecutionActor ! ExecuteJobCommand
-          Success(ExecutionStatus.Starting)
-        case Failure(reason) =>
-          log.error(s"Failed to fetch locally qualified inputs for call ${call.fullyQualifiedName}", reason)
-          throw new WorkflowExecutionException(List(reason))
-      }
+    // FIXME This is a potential bottleneck as input evaluation can be arbitrarily long and executes code coming from the backend.
+    // It should probably be actorified / isolated
```

Comment:
Since the process of running a job is looking a lot like a self-contained state machine (i.e. you have to init, evaluate inputs, begin executing, poll for completion, evaluate outputs) then this idea is awesome and I fully support a sub-actor!

Plus it allows this actor to worry about co-ordinating calls, while the sub-actor can handle the tedium of actually running the jobs.